### ISSUE TITLE
Remove code binaries from ShaderRD header files (*.glsl.gen.h)

### ISSUE THREAD

**Issue by sanseiuP** [godotengine/godot-proposals#10782]
### Describe the project you are working on

A demo project for testing godot render features

### Describe the problem or limitation you are having in your project

When I was debugging shaders, I noticed that some changes were made to auto generated header files (*.glsl.gen.h) after I modify the source glsl file. This results in some C++ code recompilation, which is a little bit annoying. It seems because glsl source code are translated to hard coded char array in a header file. 
![image](https://github.com/user-attachments/assets/d73366e3-c52c-4ab1-8eaa-c3b7cc5a908f)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I think it would be better to store the binaries to a separate file, or simply load from source text at engine launch time. So it won't make C++ to recompile. Also, this makes it possible to reload shaders without restarting engine, which makes graphic programmers happy.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Generate binaries to another place. Use get member functions in ShaderRD to load shader code from disk. So header files won't be changed.
![image](https://github.com/user-attachments/assets/34ecf67e-f8df-4302-914e-d8c103b26574)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No. It makes a lot of change. But It makes development more efficient in the long run.

### Is there a reason why this should be core and not an add-on in the asset library?

It's about core designs.

**Comment by sanseiuP** [comment]
> This would be really nice to have for iterating on the core shaders in dev builds. Its not something that we should include outside of dev builds

I agree. There can be an option to load shaders directly from source code in dev builds.

**Comment by stuartcarnie** [comment]
@clayjohn do we want to specify changes to how it should work? We could consider generating a new file that generates all the `GLOBAL_DEF`s for the built-in shaders?
